specification/storage/resource-manager/Microsoft.Storage/stable/2023-05-01/common.json

```diff
@@ -0,0 +1,362 @@
+          "$ref": "#/definitions/CloudErrorBody"
+        }
+      },
+      "description": "An error response from the Storage service."
+    },
+    "CloudErrorBody": {
+      "x-ms-external": true,
+      "properties": {
+        "code": {
+          "type": "string",
+          "description": "An identifier for the error. Codes are invariant and are intended to be consumed programmatically."
+        },
+        "message": {
+          "type": "string",
+          "description": "A message describing the error, intended to be suitable for display in a user interface."
+        },
+        "target": {
+          "type": "string",
+          "description": "The target of the particular error. For example, the name of the property in error."
+        },
+        "details": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/CloudErrorBody"
+          },
+          "description": "A list of additional details about the error."
+        }
+      },
+      "description": "An error response from the Storage service."
+    },
+    "SignedIdentifier": {
+      "properties": {
+        "id": {
+          "type": "string",
+          "description": "An unique identifier of the stored access policy."
+        },
+        "accessPolicy": {
+          "$ref": "#/definitions/AccessPolicy",
+          "description": "Access policy"
+        }
+      }
+    },
+    "AccessPolicy": {
+      "properties": {
+        "startTime": {
+          "type": "string",
+          "format": "date-time",
+          "description": "Start time of the access policy"
+        },
+        "expiryTime": {
+          "type": "string",
+          "format": "date-time",
+          "description": "Expiry time of the access policy"
+        },
+        "permission": {
+          "type": "string",
+          "description": "List of abbreviated permissions."
+        }
+      }
+    },
+    "StorageTaskReportSummary": {
+      "type": "object",
+      "properties": {
+        "value": {
+          "readOnly": true,
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/StorageTaskReportInstance"
+          },
+          "x-ms-identifiers": [],
+          "description": "Gets storage tasks run result summary."
+        },
+        "nextLink": {
+          "type": "string",
+          "description": "Request URL that can be used to query next page of storage task run results summary. Returned when the number of run instances and summary reports exceed maximum page size.",
+          "readOnly": true
+        }
+      },
+      "description": "Fetch Storage Tasks Run Summary."
+    },
+    "StorageTaskReportInstance": {
+      "type": "object",
+      "properties": {
+        "properties": {
+          "$ref": "#/definitions/StorageTaskReportProperties"
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "../../../../../common-types/resource-management/v5/types.json#/definitions/ProxyResource"
```

Comment:
>   "$ref": "../../../../../common-types/resource-management/v5/types.json#/definitions/ProxyResource" [](http://example.com/codeflow?start=7&length=101)

are these report instances, actually resources with their own resource id and everything?

Reply:
Yes they do have an Id with server generated name, this was suggested by ARM when we did the same for Microsoft.StorageActions swagger
https://github.com/Azure/azure-rest-api-specs/pull/28461#discussion_r1573045656